File: Depth-first_search(DFS)/python/01_바이러스.py
# fast campus 강의
# https://www.acmicpc.net/problem/2606
# 1

n = int(input())
m = int(input())
adj = [[] for _ in range(n + 1)]
count = 0

for i in range(m):
    start, end = map(int, input().split())
    adj[start].append(end)
    adj[end].append(start)

def dfs(v):
    global count
    visited[v] = True
    for e in adj[v]:
        if not(visited[e]):
            count += 1
            dfs(e)

visited = [False] * (n + 1)
dfs(1)

print(count)

File: Depth-first_search(DFS)/c++/01_바이러스.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/2606
// 1

#include <iostream>
#include <vector>

using namespace std;

vector<int> virus[101];
bool visited[101];
int N, M, count;

void dfs(int x)
{
    visited[x] = true;
    for (int i = 0; i < virus[x].size(); i++) {
        int y = virus[x][i];
        if (!visited[y]) {
            count++;
            dfs(y);
        }
    }
}

int main()
{
    cin >> N >> M;
    for (int i = 0; i < M; i++) {
        int x, y;
        cin >> x >> y;
        virus[x].push_back(y);
        virus[y].push_back(x);
    }

    dfs(1);
    cout << count;
}